### ISSUE TITLE
error: Invalid JSON PatchKaniko

### ISSUE THREAD

**Issue by andrey-gava** [kvaps/kubectl-build#12]
`kubectl build -c . -d nexus.domain.com:445/portal-test:692 --namespace default`

Output:
```
spawning "kaniko-drlmgw"
error: Invalid JSON PatchKaniko 102400 (100KiB, 63MiB/s)
tar: -: Записано только 94208 байт из 102400                                                                                                                                                  
tar: Child returned status 141                                                                                                                                                                
tar: Error is not recoverable: exiting now
```

With set x
```
+ tar --version
+ awk {print $1; exit}
+ m=tar
+ [ tar != bsdtar ]
+ [ -f /home/user/.docker/config.json ]
+ tar -P --transform s|^/home/user/.docker/config.json|../.docker/config.json| --record-size=100K --checkpoint=1 --checkpoint-action=ttyout=Sending build context to Kaniko %{r}T\r --exclude-ignore=.dockerignore -C . -czf - . /home/user/.docker/config.json
error: Invalid JSON PatchKaniko 102400 (100KiB, 48MiB/s)
+ ec=1
+ [ false = true ]
+ exit 1
```

**Comment by kvaps** [comment]
Hi, thanks for the report, could you provide full log of execution with `-x` enabled.
You can paste it inside the special spoiler tag: 

```html
<details><summary>command output</summary>
paste your output here
</details>
```

**Comment by andrey-gava** [comment]
<details><summary>command output</summary>

```
+ export DOCKER_CONFIG=/home/gava/.docker/config.json
+ export KUBECONFIG=/tmp/kubie-confighRzvU9.yaml
+ kubectl=kubectl
+ version=1.6.0
+ image=ghcr.io/kvaps/kaniko-executor:v1.6.0
+ env LC_ALL=C tr -dc a-z0-9
+ head -c 6
+ name=kaniko-qfl2m2
+ context=
+ debug=false
+ tty=false
+ usetar=false
+ generator=
+ digestfile=
+ imagenamewithdigestfile=
+ nodefaultctx=0
+ nodefaultns=0
+ args=[ 
+ [ 7 -gt 0 ]
+ key=-c
+ context=.
+ shift
+ shift
+ [ 5 -gt 0 ]
+ key=-d
+ args=[ "-d", 
+ shift
+ [ 4 -gt 0 ]
+ key=nexus.domain.com:445/portal-smm-ws-wwcc:692
+ args=[ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", 
+ shift
+ [ 3 -gt 0 ]
+ key=--namespace
+ nodefaultns=1
+ kubectl=kubectl --namespace default
+ shift
+ shift
+ [ 1 -gt 0 ]
+ key=--dockerfile=Dockerfile333
+ args=[ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", "--dockerfile=Dockerfile333", 
+ shift
+ [ 0 -gt 0 ]
+ [ 0 = 1 ]
+ kubectl config current-context
+ kubectl=kubectl --namespace default --context=portal
+ [ 1 = 1 ]
+ [ -n  ]
+ [ -n  ]
+ args=[ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", "--dockerfile=Dockerfile333", "--image-name-with-digest-file="
+ [ -d . ]
+ args=[ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", "--dockerfile=Dockerfile333", "--image-name-with-digest-file=", "--context=tar://stdin"
+ usetar=true
+ tty=false
+ [ false = true ]
+ args=[ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", "--dockerfile=Dockerfile333", "--image-name-with-digest-file=", "--context=tar://stdin" ]
+ cat
+ overrides={
  "metadata": {"},
  "spec": {
    "containers": [
      {
        "image": "ghcr.io/kvaps/kaniko-executor:v1.6.0",
        "name": "kaniko",
        "tty": false,
        "stdin": true,
        "stdinOnce": true,
        "terminationMessagePath": "/dev/termination-log",
        "terminationMessagePolicy": "File",
        "args": [ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", "--dockerfile=Dockerfile333", "--image-name-with-digest-file=", "--context=tar://stdin" ]
      }
    ],
    "automountServiceAccountToken": false,
    "enableServiceLinks": false
  }
}
+ kubectl version --client -o yaml
+ awk -F[ :"]+ $2 == "minor" {print $3+0}
+ m=20
+ [ 20 -lt 18 ]
+ [  != true ]
+ trap ec=$?; kubectl --namespace default --context=portal delete pod kaniko-qfl2m2 --wait=false 2>/dev/null || true; exit $ec EXIT INT TERM
+ echo spawning "kaniko-qfl2m2"
spawning "kaniko-qfl2m2"
+ [ true = true ]
+ mktemp -u
+ pidfile=/tmp/tmp.DW4wtLBeD5
+ tarf -C . -czf - .
+ kubectl --namespace default --context=portal run --image ghcr.io/kvaps/kaniko-executor:v1.6.0 --restart=Never --overrides={
  "metadata": {"},
  "spec": {
    "containers": [
      {
        "image": "ghcr.io/kvaps/kaniko-executor:v1.6.0",
        "name": "kaniko",
        "tty": false,
        "stdin": true,
        "stdinOnce": true,
        "terminationMessagePath": "/dev/termination-log",
        "terminationMessagePolicy": "File",
        "args": [ "-d", "nexus.domain.com:445/portal-smm-ws-wwcc:692", "--dockerfile=Dockerfile333", "--image-name-with-digest-file=", "--context=tar://stdin" ]
      }
    ],
    "automountServiceAccountToken": false,
    "enableServiceLinks": false
  }
} -i kaniko-qfl2m2
+ tar --version
+ awk {print $1; exit}
+ m=tar
+ [ tar != bsdtar ]
+ [ -f /home/gava/.docker/config.json ]
+ tar -P --transform s|^/home/gava/.docker/config.json|../.docker/config.json| --record-size=100K --checkpoint=1 --checkpoint-action=ttyout=Sending build context to Kaniko %{r}T\r --exclude-ignore=.dockerignore -C . -czf - . /home/gava/.docker/config.json
error: Invalid JSON PatchKaniko 102400 (100KiB, 79MiB/s)
+ ec=1
+ [ false = true ]
+ exit 1
tar: -: Записано только 94208 байт из 102400                                                                                                                                                  
tar: Child returned status 141                                                                                                                                                                
tar: Error is not recoverable: exiting now
+ ec=1
+ kubectl --namespace default --context=portal delete pod kaniko-qfl2m2 --wait=false
+ true
+ exit 1
```

</details>

**Comment by kvaps** [comment]
Thanks for output, should be fixed in `v1.6.1`
https://github.com/kvaps/kubectl-build/commit/f75fb193a9d49a7ea647151107e9ffb44a913422

related problem described here
https://unix.stackexchange.com/questions/401728/how-do-i-echo-an-empty-json-curly-brackets-as-a-default-value
